packages/core/upload/admin/src/pages/App/ConfigureTheView/state/reducer.js

```diff
@@ -0,0 +1,29 @@
+import produce from 'immer'; // current
+import set from 'lodash/set';
+import get from 'lodash/get';
+import { ON_CHANGE, SET_LOADED } from './actionTypes';
+
+const initialState = {
+  initialData: {},
+  modifiedData: {},
+};
+
+const reducer = (state = initialState, action) =>
+  // eslint-disable-next-line consistent-return
+  produce(state, (draftState) => {
+    switch (action.type) {
+      case ON_CHANGE: {
+        set(draftState, ['modifiedData', ...action.keys.split('.')], action.value);
+        break;
+      }
+      case SET_LOADED: {
+        set(draftState, ['initialData'], get(draftState, ['modifiedData'], {}));
```

Comment:
I'm struggling to understand the point of this action?

Reply:
This action is reinitialising the state in order to disable the save button. Maybe this is easier to understand?

```
        const reInitialise = init(get(draftState, ['modifiedData'], {}));
        draftState.initialData = reInitialise.initialData;
        draftState.modifiedData = reInitialise.modifiedData;
        break;
```

Reply:
that is easier to understand, but equally, a comment could clear it all up! 😄